ballerina/utils.bal

```diff
@@ -15,21 +15,154 @@
 // under the License.
 
 import ballerina/jballerina.java;
+import ballerina/time;
+import ballerina/regex;
+import ballerina/lang.'int as ints;
+
+isolated string[] digits = [
+    "0", "1", "2", "3", "4", "5", 
+    "6", "7", "8", "9", "a", "b", 
+    "c", "d", "e", "f", "g", "h", 
+    "i", "j", "k", "l", "m", "n", 
+    "o", "p", "q", "r", "s", "t", 
+    "u", "v", "w", "x", "y", "z"
+];
 
 isolated function getRandomUUID() returns handle = @java:Method {
     name: "randomUUID",
     'class: "java.util.UUID"
 } external;
 
-isolated function getBytesFromUUID(string uuid) returns byte[] = @java:Method {
-    name: "getBytesFromUUID",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+isolated function getBytesFromUUID(string uuid) returns byte[]|Error {
+    int msb = check getMostSignificantBits(uuid);
+    int lsb = check getLeastSignificantBits(uuid);
+    return getBytesFromSignificantBits(msb, lsb);
+}
 
-isolated function getUUIDFromBytes(byte[] uuid) returns string = @java:Method {
-    name: "getUUIDFromBytes",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+isolated function getUUIDFromBytes(byte[] uuid) returns string {
+    int msb = ((uuid[0] & 0xFF) << 56) |
+            ((uuid[1] & 0xFF) << 48) |
+            ((uuid[2] & 0xFF) << 40) |
+            ((uuid[3] & 0xFF) << 32) |
+            ((uuid[4] & 0xFF) << 24) |
+            ((uuid[5] & 0xFF) << 16) |
+            ((uuid[6] & 0xFF) << 8) |
+            ((uuid[7] & 0xFF) << 0);
+
+    int lsb = ((uuid[8] & 0xFF) << 56) |
+            ((uuid[9] & 0xFF) << 48) |
+            ((uuid[10] & 0xFF) << 40) |
+            ((uuid[11] & 0xFF) << 32) |
+            ((uuid[12] & 0xFF) << 24) |
+            ((uuid[13] & 0xFF) << 16) |
+            ((uuid[14] & 0xFF) << 8) |
+            ((uuid[15] & 0xFF) << 0);
+
+    return getUuidFromSignificantBits(msb, lsb);
+}
+
+isolated function getMostSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
```

Comment:
Since this `toRecord()` is called both MSB and LSB logic. shall we convert it once and pass the Uuid record?

Reply:
Fixed in [6d52d6d](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/6d52d6da9c8efa6d0a0f561cbc289fc9af21f1e9)